### PR TITLE
[export] Implement cpp deserializer.

### PR DESCRIPTION
**Pull request by zhxchen17**
Differential Revision: D63206258

This diff introduces a mechanism to generate a json-compatible deserializer in cpp using nlohmann json (already being used by AOTI). 

Why we need this? Because there will be a lot of cases where people don't want to use Python to load the graph (e.g. cpp runtime), and instead they can use this header to deserialize the JSON graph.

Every time we call update_schema.py to update the schema, the header will be auto generated and included into the source files.

### CODE REVIEW

**Review by seemethere — CHANGES_REQUESTED** [review]
You should add the generated file to the list of generated files here:

https://github.com/pytorch/pytorch/blob/main/.gitattributes

Then the check should pass since it ignores generated files.

File: torch/_export/serde/serialize.py [hunk 1]
```diff
@@ -770,13 +770,13 @@ def serialize_input(
             # For regular FX graph, SymInt arg should be a fx.Node with
             # self.is_sym_int_arg(arg) being true
             return Argument.create(as_sym_int=SymIntArgument.create(as_name=str(arg)))
-        elif isinstance(arg, bool):
+        elif type(arg) is bool:
```
[the remark this answers is not in the document]
Comment (zhxchen17) [reply torch/_export/serde/serialize.py:773]: I think it's for keeping consistency with isinstance(arg, int). no difference I know of, but instance(arg, bool) might have surprises we don't know, like int.

File: torch/_C/_aoti.pyi [hunk 2]
```diff
@@ -16,6 +16,15 @@ def alloc_tensor_by_stealing_from_void_ptr(
     handle: c_void_p,
 ) -> Tensor: ...
 
+class CppExportedProgram: ...
```
[the remark this answers is not in the document]
Comment (zhxchen17) [reply torch/_C/_aoti.pyi:19]: oki

**Review by angelayi — APPROVED** [review]

**Pull request closed without merging**